tree/ntuple/v7/test/ntuple_view.cxx

```diff
@@ -258,3 +258,125 @@ TEST(RNTuple, MissingViewNames)
       EXPECT_THAT(err.what(), testing::HasSubstr("no field named 'badC' in RNTuple 'myNTuple'"));
    }
 }
+
+class RNTupleViewExternalAddressTests : public ::testing::TestWithParam<std::pair<bool, bool>> {};
+
+TEST_P(RNTupleViewExternalAddressTests, ViewExternal)
+{
+   FileRaii fileGuard("test_ntuple_viewexternal.root");
+
+   auto model = RNTupleModel::Create();
+   auto fieldPt = model->MakeField<float>("pt", 42.0);
+
+   {
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      writer->Fill();
+   }
+
+   auto reader = RNTupleReader::Open("ntpl", fileGuard.GetPath());
+
+   auto [typed, use_shared_ptr] = GetParam();
+   // Cover the four cases here
+   if (typed && use_shared_ptr) {
+      auto mydata = std::make_shared<float>();
+      auto viewPt = reader->GetView("pt", mydata);
+      viewPt(0);
+      EXPECT_FLOAT_EQ(42.0, *mydata);
+   } else if (!typed && !use_shared_ptr) {
+      float mydata;
+      auto viewPt = reader->GetView("pt", reinterpret_cast<void *>(&mydata));
+      viewPt(0);
+      EXPECT_FLOAT_EQ(42.0, mydata);
+   } else if (typed && !use_shared_ptr) {
+      float mydata;
+      auto viewPt = reader->GetView("pt", &mydata);
+      viewPt(0);
+      EXPECT_FLOAT_EQ(42.0, mydata);
+   } else if (!typed && use_shared_ptr) {
+      std::shared_ptr<void> mydata{new float()};
+      auto viewPt = reader->GetView("pt", mydata);
+      viewPt(0);
+      EXPECT_FLOAT_EQ(42.0, *static_cast<float *>(mydata.get()));
+   }
```

Comment:
What is the advantage of templating the test? Can you just spell out the four checks one after the other, in the same test function?

Reply:
Indeed! I guess I was feeling fancy in that moment